### PR TITLE
feat: stubs declaring strict types 

### PR DESCRIPTION
**Pull request by juantejer4**
# ⚡ Update Laravel boilerplate's stubs to declare strict types - [task](https://app.clickup.com/t/86dr22xuk) ⚡

## 💻 What type of change is this?

- [x] 💎 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Styling
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build / CI

## ⭐ Description

All stubs, including those for controllers, requests, and other artisan-generated files, have been updated to include declare(strict_types=1) at the top. This change will apply to all new code generated using artisan commands, promoting consistency and reliability in type usage throughout the application.

## 📸 Evidence 
Now you can run for example `sail artisan make:controller TestController`
This process will confirm whether the modified stubs are being used for generating new files with the strict types declaration.
image

## ✅ Checklist

- [x] This PR can be merged (it is not a draft, work in progress, or blocked on another PR)
- [ ] I have tested this change locally in multiple screen sizes
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: cspell.json [hunk 1]
```diff
@@ -11,6 +11,9 @@
     "vendor"
   ],
   "words": [
+    "Dispatchable",
+    "Queueable",
+    "Mailables",
```
Comment (gianfranco-rocco) [inline cspell.json:16]: Sort words ascending?

File: stubs/cast.inbound.stub [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/cast.inbound.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/cast.stub [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/cast.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/console.stub [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/console.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/controller.api.stub [hunk 5]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/controller.api.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/controller.invokable.stub [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/controller.invokable.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/controller.model.api.stub [hunk 7]
```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/controller.model.api.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/controller.model.stub [hunk 8]
```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/controller.model.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/controller.nested.api.stub [hunk 9]
```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/controller.nested.api.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/controller.nested.singleton.api.stub [hunk 10]
```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/controller.nested.singleton.api.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/controller.nested.singleton.stub [hunk 11]
```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/controller.nested.singleton.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/controller.nested.stub [hunk 12]
```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/controller.nested.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/controller.plain.stub [hunk 13]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/controller.plain.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/controller.singleton.api.stub [hunk 14]
```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/controller.singleton.api.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/controller.singleton.stub [hunk 15]
```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/controller.singleton.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/controller.stub [hunk 16]
```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/controller.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/event.stub [hunk 17]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/event.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/factory.stub [hunk 18]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/factory.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/job.queued.stub [hunk 19]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/job.queued.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/job.stub [hunk 20]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/job.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/mail.stub [hunk 21]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/mail.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/markdown-mail.stub [hunk 22]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/markdown-mail.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/markdown-notification.stub [hunk 23]
```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/markdown-notification.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/middleware.stub [hunk 24]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/middleware.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/migration.create.stub [hunk 25]
```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/migration.create.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/migration.stub [hunk 26]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/migration.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/migration.update.stub [hunk 27]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/migration.update.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/model.pivot.stub [hunk 28]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/model.pivot.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/model.stub [hunk 29]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/model.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/notification.stub [hunk 30]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/notification.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/observer.plain.stub [hunk 31]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/observer.plain.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/observer.stub [hunk 32]
```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/observer.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/policy.plain.stub [hunk 33]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/policy.plain.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/policy.stub [hunk 34]
```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/policy.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/provider.stub [hunk 35]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/provider.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/request.stub [hunk 36]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/request.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/resource-collection.stub [hunk 37]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/resource-collection.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/resource.stub [hunk 38]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/resource.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/rule.stub [hunk 39]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/rule.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/scope.stub [hunk 40]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/scope.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/seeder.stub [hunk 41]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/seeder.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/test.stub [hunk 42]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/test.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/test.unit.stub [hunk 43]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/test.unit.stub:3]:
```suggestion
declare(strict_types=1);
```

File: stubs/view-component.stub [hunk 44]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1)
```
Comment (gianfranco-rocco) [inline stubs/view-component.stub:3]:
```suggestion
declare(strict_types=1);
```

**Review by gianfranco-rocco — CHANGES_REQUESTED** [review]
Did you run some commands to validate they're generated with the strict types declaration?

**Comment by juantejer4** [comment]
> Did you run some commands to validate they're generated with the strict types declaration?

Yep, for instance running `sail artisan make:controller TestController`:
image

**Review by gianfranco-rocco — APPROVED** [review]

**Pull request merged**